service/gcs/runtime/runc/utils.go

```diff
@@ -88,3 +92,40 @@ func (r *runcRuntime) processExists(pid int) bool {
 	_, err := os.Stat(filepath.Join("/proc", strconv.Itoa(pid)))
 	return !os.IsNotExist(err)
 }
+
+func getRuncLogError(logPath string) interface{} {
+	reader, err := os.OpenFile(logPath, syscall.O_RDONLY, 0644)
+	if err != nil {
+		return nil
+	}
+	defer reader.Close()
+
+	var lastErr interface{}
+	dec := json.NewDecoder(reader)
+	for {
+		var out map[string]interface{}
```

Comment:
It's interesting to me that we're unmarshaling the JSON into a map instead of into a struct like we did in hcsshim when handling the same case... Is there a reason for this?

Reply:
The json decoder used here doesn't differentiate between an error that is the result of the stream being broken and an error from unmarshalling. However, I'm not sure that matters too much since in the event the logs aren't returned to the user on error, we can still manually check the logs in the uvm. I'll update it.